src/input.jl

```diff
@@ -0,0 +1,96 @@
+
+struct GenericLinearObservation <: AbstractObservation
+end
+struct GenericLinearEvolution{TPhi,Tb,TQ} <: AbstractEvolution
+end
+
+"""
+    LinearEvolution(Phi, b, Q)
+
+Evolution `x -> Phi x + b + w` where ``w ~ N(0, Q)``
+"""
+struct LinearEvolution{TPhi,Tb,TQ} <: AbstractEvolution
+    Phi::TPhi # dxd
+    b::Tb # d
+    Q::TQ # dxd
+end
+
+"""
+    LinearObservation(H, R)
+
+Observe `y = Hx + v` where ``v ~ N(0, R)``
+"""
+struct LinearObservation{TH, TR} <: AbstractObservation
+    H::TH # dxd
+    R::TR # d
+end
+
+
+function predict!(s, x, P, t, u, M) 
+    if s == t # if `s == t`` no time has passed
```

Comment:
Shouldn't this special-casing also be done in `predict!(s, x, P, t, U, M::GenericLinearEvolution)`?